docs/src/trace-viewer-intro-csharp-java-python.md

```diff
@@ -0,0 +1,120 @@
+---
+id: trace-viewer-intro
+title: "Trace Viewer"
+---
+
+Playwright Trace Viewer is a GUI tool that lets you explore recorded Playwright traces of your tests meaning you can go back and forward though each action of your test and visually see what was happening during each action.
+
+**You will learn**
+
+- How to record a trace
+- How to open the HTML report
+- How to open the trace viewer
+
+## Recording a trace
+
+Traces can be recorded using the [`property: BrowserContext.tracing`] API as follows:
+
+```java
+Browser browser = browserType.launch();
+BrowserContext context = browser.newContext();
+
+// Start tracing before creating / navigating a page.
+context.tracing().start(new Tracing.StartOptions()
+  .setScreenshots(true)
+  .setSnapshots(true)
+  .setSources(true));
+
+Page page = context.newPage();
+page.navigate("https://playwright.dev");
+
+// Stop tracing and export it into a zip archive.
+context.tracing().stop(new Tracing.StopOptions()
+  .setPath(Paths.get("trace.zip")));
+```
+
+```python async
+browser = await chromium.launch()
+context = await browser.new_context()
+
+# Start tracing before creating / navigating a page.
+await context.tracing.start(screenshots=True, snapshots=True, sources=True)
```

Comment:
We should mention the pytest plugin way there instead of the library approach.

Reply:
you will have to add what you mean here as no idea what the pytest plugin way is

Reply:
https://playwright.dev/python/docs/test-runners

`pytest --trace on`